docs/api-ref.md

```diff
@@ -1645,6 +1640,425 @@ Error  |  Description
+# server = TSC.Server('https://MY-SERVER')  
+# sign in, etc.  
+
+ all_project_items, pagination_item = server.projects.get()
+ project_item = all_project_items[0]
+ server.projects.populate_workbook_default_permissions(project_item)
+``` 
+
+#### projects.populate_datasource_default_permissions
+```py
+projects.populate_datasource_default_permissions(item)
+```
+
+Queries the default datasource permissions, parses and stores the returned the permissions.
+
+REST API: [Query Default Permissions](https://help.tableau.com/current/api/rest_api/en-us/REST/rest_api_ref_permissions.htm#query_default_permissions){:target="_blank"}  
+
+**Parameters**
+
+Name  |  Description  
+:--- | :---  
+`item`   | A project object.
+
+**Example**  
+ 
+```py
+# import tableauserverclient as TSC  
+# server = TSC.Server('https://MY-SERVER')  
+# sign in, etc.  
+
+ all_project_items, pagination_item = server.projects.get()
+ project_item = all_project_items[0]
+ server.projects.populate_datasource_default_permissions(project_item)
+``` 
+
+
+#### projects.populate_flow_default_permissions
+```py
+projects.populate_flow_default_permissions(item)
+```
+
+Queries the default flow permissions, parses and stores the returned the permissions.
+
+REST API: [Query Default Permissions](https://help.tableau.com/current/api/rest_api/en-us/REST/rest_api_ref_permissions.htm#query_default_permissions){:target="_blank"}  
+
+
+**Parameters**
+
+Name  |  Description  
+:--- | :---  
+`item`   | A project object.
+
+**Example**  
+ 
+```py
+# import tableauserverclient as TSC  
+# server = TSC.Server('https://MY-SERVER')  
+# sign in, etc.  
+
+ all_project_items, pagination_item = server.projects.get()
+ project_item = all_project_items[0]
+ server.projects.populate_flow_default_permissions(project_item)
+``` 
+
+#### projects.update_workbook_default_permissions
+```py
+projects.update_workbook_default_permissions(item, rules)
+```
+
+Add or update default workbook permissions.
+
+REST API: [Add Default Workbook Permissions](https://help.tableau.com/current/api/rest_api/en-us/REST/rest_api_ref_permissions.htm#add_default_permissions){:target="_blank"}  
+
+**Parameters**
+
+Name  |  Description  
+:--- | :---  
+`item`   | A project object.
+`rules`   | A list of PermissionsRule objects.
+
+**Example**  
+ 
+```py
+# import tableauserverclient as TSC  
+# server = TSC.Server('https://MY-SERVER')  
+# sign in, etc.  
+
+ all_project_items, pagination_item = server.projects.get()
+ project_item = all_project_items[0]
+ 
+ capabilities = {
+   TSC.Permission.Capability.ViewComments: TSC.Permission.Mode.Allow
+ }
+
+ rules = TSC.PermissionsRule(
+  grantee=project_item,
+  capabilities=capabilities
+ )
+ 
+ server.projects.update_workbook_default_permissions(project_item, [rules])
+``` 
+
+#### projects.update_datasource_default_permissions
+```py
+projects.update_datasource_default_permissions(item, rules)
+```
+
+Add or update default datasource permissions.
```

Comment:
You get the idea:

Add or update default permissions for data sources in a project.